src/wasm/wasm.cpp

```diff
@@ -1526,10 +1527,16 @@ void Resume::finalize() {
   if (handleUnreachableOperands(this)) {
     return;
   }
+  if (cont->type.isNull()) {
+    // This will never be executed and the instruction will not be emitted.
+    // Model this with an uninhabitable cast type.
+    type = cont->type.with(NonNullable);
```

Comment:
`Resume` does not return a continuation, so this is not correct. It should return the result types given by the resumed continuation.

Reply:
Agreed this is wrong, but perhaps I can leave it to a later followup that gets the upstream spec test passing? All this will be tested at that time fully, I think. (Changes in this PR are just to get some amount of spec tests passing, enough to get to interesting parts.)

Reply:
It's difficult to tell whether things are intentionally or unintentionally wrong :) Maybe it would be worth getting validation working earlier rather than later? We should be able to validate the spec tests before we can execute them.

Reply:
Hmm, the thing is that the spec test is nor ordered in such a way... which would have been nice.

I am basically just going top-down in that test, to make sure I don't miss anything.

I promise we will pass it all after all followups land! :smile:

Reply:
Coming back to this TODO, I'm not sure how to fix it. Yes, this should return the result of the continuation, but there is no continuation type here - it is null - so what can we do aside from emit an uninhabitable type?

Reply:
Ok, opened https://github.com/WebAssembly/binaryen/pull/7794 for that.